dub.sdl

```diff
@@ -14,31 +14,31 @@ buildType "unittest" {
     versions "mir_bignum_test" "mir_bignum_test_llv"
     // versions "mir_ndslice_test"
     versions "mir_test"
-    dflags "-lowmem"
+    dflags "-lowmem" "-checkaction=context" "-allinst"
```

Comment:
Because otherwise ilnking fails with -checkaction=context.

Reply:
It's because prebuilt druntime and Phobos aren't compiled with `-checkaction=context`. So when the compiler comes across some `assert(x == y)` in an imported druntime/Phobos file, which is lowered to a druntime template instantiation (with many child instantiations) with `-checkaction=context`, it tries not to re-codegen instantiations of that subtree by default. `-allinst` is supposed to emit all template instantiations from all root modules being compiled (but into a single object file only), making the resulting set of compiled object files self-sufficient wrt. all instantiated symbols (so no checkaction-related symbols needed from druntime/Phobos). It doesn't work 100%, but it seems to be good enough for most use cases now. Just don't use it when compiling each module separately, that likely results in a big compilation slow-down.

Reply:
Thanks!

Reply:
Why isn't druntime and phobos compiled with `-checkaction=context`? Because it increase binary size?

Reply:
IIRC, the default druntime/Phobos libs are compiled with `-release`, so assertions are disabled in general and `-checkaction=context` wouldn't make a difference anyway. It would for LDC's debug libs though.

It's a general problem whenever code is linked together that was compiled with different flags - unless using `-allinst` (or LDC's `-linkonce-templates`).

Reply:
> IIRC, the default druntime/Phobos libs are compiled with -release, so assertions are disabled in general and -checkaction=context wouldn't make a difference anyway.

But would avoid having to use `-allinst` when using `-checkaction=context` when building packages?

Reply:
Nope - the compiler would again assume that druntime and Phobos have been compiled with the exact same flags as the current compilation, i.e., assume assertions were generally enabled, not just for `@safe` functions.